src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
             }
         }
 
+        public async Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync()
```

Comment:
Method declaration has clear message about the purpose of tuple and it's values. Whoever consume this API, will have clear understanding of the return values. I find it more convenient and easy to understand than having multiple out params and that's the whole of point of having tuples.